frontend/src/board/board.ts

```diff
@@ -45,5 +47,105 @@ export class Board {
       { row: 0, column: 2},
       { row: 0, column: 3},
     ];
+
+  }
+
+  findMoves() {
+    return {
+      north: this.checkDirections(this.robotPositions[0].row, this.robotPositions[0].column, 'North'),
+    south: this.checkDirections(this.robotPositions[0].row, this.robotPositions[0].column, 'South'),
+    east: this.checkDirections(this.robotPositions[0].row, this.robotPositions[0].column, 'East'),
+    west: this.checkDirections(this.robotPositions[0].row, this.robotPositions[0].column, 'West')
+    }
+  }
+
+  checkDirections(startRow:number, startCol: number, direction: 'North' | 'South' | 'East' | 'West' ) : null | Position {
+    // Gets the position of the target robot
+    // Checks the north, south, east, and west path
+    // Path ends if
+    // - it is the last cell in the direction
+    // - or the cell is obstructed with either another robot, a wall, or the center pieces
+    // - or the cell is the target cell
+
+    let legalMove: Position | null = null 
+    
+    
+    let row = startRow
+    let col = startCol
+ 
+    while (row >= 0 && row < BOARD_SIZE && col >= 0 && col < BOARD_SIZE) {
+
+      // check if target, add position
+
+      let robotEncountered = false 
+      // robots 
+      this.robotPositions.forEach(robot => {
+        // designated robot does not check its own position
+        if (robot.row === startRow && robot.column === startCol) {
+          return 
```

Comment:
This check is probably unnecessary - there should only be on robot per square (and if there isn't we probably have something we need to fix somewhere else)